### ISSUE TITLE
Training Speed Problem

### ISSUE THREAD

**Issue by 0three** [amazon-research/gan-control#5]
Hi, I've tried to reproduce the training-generator phase. 

However, the training speed seems too slow. 
![image](https://user-images.githubusercontent.com/20237367/165722712-f4f0be00-de40-43e4-aeca-9fd2920d9269.png)

I use 4 A100 for default training. It takes almost 4 minutes for 4 iterations.  4 GPUs are all 100% GPU-utils.

**Comment by 0three** [comment]
![image](https://user-images.githubusercontent.com/20237367/165724200-3ad4f649-3f7a-4150-a258-dad99385c649.png)

**Comment by 0three** [comment]
I checked the whole training process . The func generator_regularize_step will cost 200s each time, which greatly damage the reproducity.

**Comment by 0three** [comment]
I tried to leverage faster ops from rosinality's repo. It rises to 140s each time. It is still unbearable. Could you plz provide some solutions?

**Comment by AlonShoshan10** [comment]
Using the ops from rosinality, the training speed we get on 4 V100 is 1.5-1.8 sec per iteration (on average including the generator_regularize_step every 4th iter).

What is your batch size/image size?
Looking at your memory usage, its higher then 11-13GB In our training.

**Comment by 0three** [comment]
I've fixed this bug by replacing A100 to V100.

**Comment by face-competition** [comment]
@AlonShoshan10 So, it takes about 16 days for finishing the training?

**Comment by AlonShoshan10** [comment]
Yes, the first phase takes 2-3 weeks, if the FID is converged you can stop it earlier. The second phase takes a few hours.
